cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/operationrunner/ListAppsFailureException.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.sourcecontrol.operationrunner;
+
+/**
+ * Exception thrown when list operation fails in operation runner.
+ * Encapsulates some underlying exceptions.
+ */
+public class ListAppsFailureException extends Exception {
```

Comment:
Please don't introduce tons of checked exceptions without a common parent. It's really hard to handle

Reply:
Any specific reason you need a separate exception here? Ususaly exceptions are per component at max (if not per whole service) and separated by error type, not by call type

Reply:
Looks like similar suggestions was given in https://github.com/cdapio/cdap/pull/14928#discussion_r1119474835

Will do the necessary change of having a single exception type for all specific SCM failures as part of this PR.

Reply:
Based on offline discussion
Updated the PR to have a single unchecked exception for all operation runner failures. Added a TODO to revisit the errors and classify into root cause based subclass exceptions later.